### PR TITLE
[omm] Hash api implementation first draft

### PR DESCRIPTION
**Pull request by Dcallies**
Summary
---------

Use the storage interfaces to pull the currently configured SignalType / ContentTypes, then download the content from the given URL and then process it.

Test Plan
---------

```bash
curl 'localhost:5000/h/hash?'\
'url=https://github.com/facebook/ThreatExchange/blob/main/pdq/data/bridge-mods/aaa-orig.jpg?raw=true&'\
'content_type=photo'
```
Response:
```
{
  "pdq": "f8f8f0cee0f4a84f06370a22038f63f0b36e2ed596621e1d33e6b39c4e9c9b22"
}
```

### CODE REVIEW

File: open-media-match/src/OpenMediaMatch/__init__.py [hunk 1]
```diff
@@ -16,7 +18,14 @@ def create_app():
...
     app = flask.Flask(__name__)
-    app.config.from_envvar("OMM_CONFIG")
+    if "OMM_CONFIG" in os.environ:
+        app.config.from_envvar("OMM_CONFIG")
+    elif sys.argv[0].endswith("/flask"):  # Default for flask CLI
+        # The devcontainer settings. If you are using the CLI outside
+        # the devcontainer and getting an error, just override the env
+        app.config.from_pyfile("/workspace/.devcontainer/omm_config.py")
+    else:
+        raise RuntimeError("No flask config given - try populating OMM_CONFIG env")
```
Comment (dougneal) [inline open-media-match/src/OpenMediaMatch/__init__.py:28]: Why was this needed?

File: open-media-match/src/OpenMediaMatch/blueprints/hashing.py [hunk 2]
```diff
@@ -10,18 +26,65 @@ def hash_media():
     Fetch content and return its hash.
     TODO: implement
     """
+
+    content_type = _parse_request_content_type()
```
Comment (dougneal) [inline open-media-match/src/OpenMediaMatch/blueprints/hashing.py:30]: How about we get the content type from the response headers (below) and simplify the interface?
`download_resp.headers['content-type']`

**Review by dougneal — CHANGES_REQUESTED** [review]

**Review by dougneal — APPROVED** [review]

**Comment by Dcallies** [comment]
Discussed offline, doing content detection in a followup and removing as an argument for now.
